fact_extractor/plugins/unpacking/generic_carver/code/generic_carver.py

```diff
@@ -5,145 +5,100 @@
 from __future__ import annotations
 
 import logging
-import re
-import shutil
+import traceback
+from itertools import chain
 from pathlib import Path
-
-from common_helper_process import execute_shell_command
-
-from helperFunctions import magic
+from typing import Iterable
+
+import structlog
+from common_helper_unpacking_classifier import avg_entropy
+from unblob.extractor import carve_unknown_chunk, carve_valid_chunk
+from unblob.file_utils import File
+from unblob.finder import search_chunks
+from unblob.handlers import BUILTIN_HANDLERS
+from unblob.handlers.compression.zlib import ZlibHandler
+from unblob.models import Chunk, HexString, PaddingChunk, TaskResult, UnknownChunk
+from unblob.plugins import hookimpl
+from unblob.processing import Task, calculate_unknown_chunks, remove_inner_chunks
 
 NAME = 'generic_carver'
 MIME_PATTERNS = ['generic/carver']
-VERSION = '0.8'
-
-TAR_MAGIC = b'ustar'
-BZ2_EOF_MAGIC = [  # the magic string is only aligned to half bytes -> two possible strings
-    b'\x17\x72\x45\x38\x50\x90',
-    b'\x77\x24\x53\x85\x09',
-]
-REAL_SIZE_REGEX = re.compile(r'Physical Size = (\d+)')
-
-
-def unpack_function(file_path, tmp_dir):
-    """
-    file_path specifies the input file.
-    tmp_dir should be used to store the extracted files.
-    """
-
-    logging.debug(f'File Type unknown: execute binwalk on {file_path}')
-    output = execute_shell_command(f'binwalk --extract --carve --signature --directory  {tmp_dir} {file_path}')
-
-    drop_underscore_directory(tmp_dir)
-    return {'output': output, 'filter_log': ArchivesFilter(tmp_dir).remove_false_positive_archives()}
-
-
-class ArchivesFilter:
```

Comment:
Is this functionality not necessary anymore because unblob produces less / no false postitves? What about the trailing data? Would it not be sensible to keep the tests / test files?

Reply:
unblob seems to include a validity check and does a much better job at finding the end of a chunk and also does not have problems with overlapping chunks. Therefore, the filter seems unnecessary. There are some things that we have to deal with, though. Since we also carve the stuff between the valid chunks, we may need to throw things out, that aren't interesting. I already built in a "low entropy filter". A "small chunk size filter" could also make sense (I saw a carved chunk containing only 2 bytes)